### ISSUE TITLE
Doesn't work on Windows

### ISSUE THREAD

**Issue by cdelgadob** [crakjie/sbt-spark-plugin#2]
The plugin looks great, but it doesn't work on my windows machine, the error is:

```
java.io.IOException: Cannot run program "C:\spark-1.5.0-bin-hadoop2.6/bin/spark-submit": 
CreateProcess error=193, %1 no es una aplicación Win32 válida
```

My HADOOP_HOME variable is set to C:\spark-1.5.0-bin-hadoop2.6

I think it may be related to the different path delimiters, I tried to use '/', but didn't work.

**Comment by crakjie** [comment]
Mmmmm yes, so long time I haven't work on windows, I have lose right reflex.

**Comment by crakjie** [comment]
I use the object Path of sat lib. It think it's manage the sperator correctly. But I can't test on windows. So I have published a 1.0.3 which are tested on OSx and linux. Maybe you can try an close the issue if you think it's ok.

**Comment by cdelgadob** [comment]
Thanks for the quick response.

Now it's displaying the right path, but I'm getting the same error:

[error](*:submit) java.io.IOException: Cannot run program "C:\spark-1.5.0-bin-hadoop2.6\bin\spark-submit": CreateProcess error=193, %1 no es una aplicación Win32 válida

So maybe it wasn't the path, but it's more related with this issue:

http://stackoverflow.com/questions/25511402/java-io-ioexception-cannot-run-program-createprocess-error-193-1-is-not-a

If I run 'spark-submit' from the command line (I have it in my PATH variable), it runs OK.

**Comment by crakjie** [comment]
O, may be the windows version is not `C:\spark-1.5.0-bin-hadoop2.6\bin\spark-submit`but `C:\spark-1.5.0-bin-hadoop2.6\bin\spark-submit.exe`?

**Comment by crakjie** [comment]
Ok, I see it, it's a `C:\spark-1.5.0-bin-hadoop2.6\bin\spark-submit.cmd` for window.
Now I must find how to choose between unix shell and window cmd

**Comment by cdelgadob** [comment]
Maybe creating a new parameter? submitPlatform := "X" / "Win"

And defaults to "X", which is Linux/Unix/OSx

**Comment by crakjie** [comment]
It's an idea, but I prefer have automatic things. Imagine if your coworker is not on the same operating system than you it's could be boring to change this values before/after every commit.

I have publish another version I would like see you test: the `1.0.4`

**Comment by cdelgadob** [comment]
Did you publish it properly?

```
[info] Resolving com.github.crakjie#sbt-spark-plugin;1.0.4 ...
[warn]  module not found: com.github.crakjie#sbt-spark-plugin;1.0.4
[warn] ==== typesafe-ivy-releases: tried
[warn]   https://repo.typesafe.com/typesafe/ivy-releases/com.github.crakjie/sbt-
spark-plugin/scala_2.10/sbt_0.13/1.0.4/ivys/ivy.xml
[warn] ==== sbt-plugin-releases: tried
[warn]   https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.github.crak
jie/sbt-spark-plugin/scala_2.10/sbt_0.13/1.0.4/ivys/ivy.xml
[warn] ==== local: tried
[warn]   C:\Users\A590181\.ivy2\local\com.github.crakjie\sbt-spark-plugin\scala_
2.10\sbt_0.13\1.0.4\ivys\ivy.xml
[warn] ==== public: tried
[warn]   https://repo1.maven.org/maven2/com/github/crakjie/sbt-spark-plugin_2.10
_0.13/1.0.4/sbt-spark-plugin-1.0.4.pom
[info] Resolving org.scala-sbt.ivy#ivy;2.3.0-sbt-fccfbd44c9f64523b61398a0155784d
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  ::          UNRESOLVED DEPENDENCIES         ::
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  :: com.github.crakjie#sbt-spark-plugin;1.0.4: not found
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]
[warn]  Note: Some unresolved dependencies have extra attributes.  Check that th
ese dependencies exist with the requested attributes.
[warn]          com.github.crakjie:sbt-spark-plugin:1.0.4 (scalaVersion=2.10, sb
tVersion=0.13)
[warn]
[warn]  Note: Unresolved dependencies path:
[warn]          com.github.crakjie:sbt-spark-plugin:1.0.4 (scalaVersion=2.10, sb
tVersion=0.13) (C:\Users\A590181\.sbt\0.13\plugins\build.sbt#L2-3)
[warn]            +- org.scala-sbt:global-plugins:0.0 (scalaVersion=2.10, sbtVer
sion=0.13)
sbt.ResolveException: unresolved dependency: com.github.crakjie#sbt-spark-plugin
;1.0.4: not found
[...Exceptions StackTrace...]
[error] (*:update) sbt.ResolveException: unresolved dependency: com.github.crakj
ie#sbt-spark-plugin;1.0.4: not found
```

**Comment by crakjie** [comment]
It's take a little time to have this available on binary
I'm able to reach it know

**Comment by cdelgadob** [comment]
Yeah, me too :)

Now, the next issue is the command line lenght. It seems it's too long. In windows there's a limit of 8192 chararters - https://support.microsoft.com/en-us/kb/830473

Are you using absolute or relative paths for the jars? I can't see the command line.

**Comment by crakjie** [comment]
Yes because your jars are in the ivy2 folder. There might be a lot of jar. I just tested on a project the command line is 17000 char long

Did you try to add `logLevel := Level.Debug` ?

**Comment by cdelgadob** [comment]
Yeah, I can see it now.. And definitely the command line with 15.605 characters is far too long for windows.

I don't know how to solve this issue... any idea?

**Comment by crakjie** [comment]
I see two possible solution, but I can not implement this right know.
1) my favorite : create a jar with just a manifest inside, this manifest references all jar.
Like that there is just one jar in the command. Of course this jar must be generated each time there is a submit.
2) Use wildcard to shortened all jar path. But it's must be done carefully to not send un desired jar to spark. And It's will not correct this issue for huge projet.

**Comment by cdelgadob** [comment]
I prefer the first one too.

How would it work if the spark cluster is in a remote machine?

**Comment by crakjie** [comment]
I wonder too.
